sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -85,6 +86,9 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
     case p: Project =>
       projectToSQL(p, isDistinct = false)
 
+    case a @ Aggregate(_, _, e @ Expand(_, _, p: Project)) =>
```

Comment:
we can add a safe guard here, only handle it as grouping set when `p.output == p.child.output ++ a.groupingExpressions`. (`==` is wrong here as nullability may differ, we can add a `sameOutput` method that use `semanticEquals`)